packages/accordion/src/useAccordion.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { useReducer } from 'react';
+import { useSection } from './useSection';
+
+export function useAccordion({
+  expandedSections = [],
+  expandable = true,
+  collapsible = true
+} = {}) {
+  const reducer = (state, action) => {
+    const retVal = [];
+
+    action.sections.forEach(section => {
+      let expanded = false;
+
+      if (section === action.section) {
+        expanded = collapsible ? state.indexOf(section) === -1 : true;
+      } else if (expandable) {
+        expanded = state.indexOf(section) !== -1;
+      }
+
+      if (expanded) {
+        retVal.push(section);
+      }
+    });
+
+    return retVal;
+  };
+  const [state, dispatch] = useReducer(reducer, expandedSections);
+  const sections = [];
+
+  const Section = ({ section, idPrefix } = {}) => {
```

Comment:
Uppercase naming should be reserved for proper React components. Let's change this to lowercase.

Reply:
So hooks linting complains if this is neither proper React or "use...".

Reply:
Addressed these comments (online & offline) in the latest commit with `useState` vs. `useReducer`.